inst/tests/nafill.Rraw

```diff
@@ -149,8 +149,15 @@ test(4.20, colnamesInt(dt, integer()), integer())
 test(4.21, colnamesInt(dt, NULL), seq_along(dt))
 test(4.22, colnamesInt("asd", 1), error="must be data.table compatible")
 test(4.23, colnamesInt(dt, 1, check_dups="a"), error="check_dups")
+test(4.24, colnamesInt(dt, c("a", "e"), skip_absent = TRUE),as.integer(c(1,0)))
+test(4.25, colnamesInt(dt, c(1L, 4L), skip_absent = TRUE), as.integer(c(1,4)))
+test(4.26, colnamesInt(dt, c(1, 4), skip_absent = TRUE),as.integer( c(1,4)))
+test(4.27, colnamesInt(dt, c("a", NA), skip_absent = TRUE),as.integer(c(1,0)))
+test(4.28, colnamesInt(dt, c(1L, NA), skip_absent = TRUE), as.integer(c(1,NA)))
```

Comment:
It's a bit strange that `NA` returns `0L` for character input, but `NA_integer_` for numeric input.

In fact I find the current behavior odd as well:

```r
data.table:::colnamesInt(iris, NA_integer_)
# Error in data.table:::colnamesInt(iris, NA_integer_) : 
#   argument specifying columns received non-existing column(s): cols[1]=-2147483648
```

I think it's outside the scope of this PR though, I'll file a follow-up issue.

Reply:
#6070